### ISSUE TITLE
exit if command fails (with non-zero exitcode)

### ISSUE THREAD

**Issue by stuff-and-nonsense** [emcrisostomo/fswatch#221]
In a BASH script with set -e, fswatch continues to run regardless of failing commands.
To have fswatch exit if a command fails, I'm using:

fswatch . | xargs -n 1 -I % sh -c "COMMAND % || pkill fswatch"

This is not robust. Perhaps fswatch could have a --terminate option which watches exitcodes and terminates if necessary on first error? Or perhaps there is a better BASH way?

Thanks for your insight.

**Comment by emcrisostomo** [comment]
> This is not robust.

How would `fswatch` know what happened to _another_ process, to which its output has been piped to? Your expectation of its working is "not robust".

Let's take `bash` for example (http://www.gnu.org/software/bash/manual/html_node/Pipelines.html):

> Each command in a pipeline is executed in its own subshell, [...]

and

> The exit status of a pipeline is the exit status of the last command in the pipeline, unless the `pipefail` option is enabled.

**Comment by stuff-and-nonsense** [comment]
Hi Enrico,

I use the safety compliment of:

set -euxo pipefail

I've tried the following on lubuntu, and it exits (as is wanted and expected):

fswatch $(pwd) | tee filelist.txt | xargs -n 1 sh -c "exit 255"

on first trigger with:

$ touch foobar

On macOS the behaviour appears different and less useful:

If I touch a file in the current folder, I can see that filelist.txt is populated, xargs fires, but DOES NOT exit. If I touch a file again, filelist.txt is updated and xargs exits. It would appear that the second invocation of xargs is not actually run but something either checks the exit code of the previous invocation and exits if necessary or something is not getting flushed.

This behaviour causes my script to appear to hang and not exit on an error until another event arrives (which may not happen).

If I replace fswatch with seq 100000, then both OSes exit around the 13000-16000 mark. So it appears that something is afoot with it not exiting with fswatch.

Is this a known behaviour of FSEvents monitor?

I've tried replacing xargs with parallel, BASH 3 with BASH 5 and zsh, but the behaviour under macOS is the same.

Thanks for your insight and such a wonderful tool!
